### DISCUSSION TITLE
Multiple Events on Default Buttons

### DISCUSSION THREAD

**Discussion by NCR25** [artisan-roaster-scope/artisan#1428] [Q&A] [answered]
I'm attempting to run multiple event buttons when the Default buttons are run or pressed.
The buttons I created work but selecting Multiple Events 1,2,3 
Button one usually executes correctly changing a Fan setting and the slider, the second might change the slider but does not change the setting (Heater) and the third does not execute on any level. I'm attempting to control a Arduino and the button is using serial commands, like OTR;1. Can I run multiple serial command instead, not sure of the format?

Or is there a better way to do this like Alarms? Thanks.

**Comment by NCR25** [comment]
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/14091044/artisan-settings.txt)
So I created a Charge Multiple Event 7,35,24
The 7 moved the Slider and started the Fan
The 35 did not start the motor
The 24 moved the slider but did not Light the lamp.
This is a clean reset and new config.
Not sure if I'm just missing something or this is not an option?

**Comment by roasterdave** [comment, marked as answer]
Sounds like the Artisan side is doing what is expected.  Do each of the buttons 7, 35, 24 work to control the roaster when you press them manually in that order?  If yes, then it is possible the Arduino cannot process them as fast as they are being sent.  That could be solved by inserting `sleep(<float>)` actions between the multiple event button actions.

**Comment by NCR25** [comment]
I was going to try the sleep option but did not think of the issue as timing?
Thank you roasterdave, that fixed the issue. Can I use sleep on other commands like Serial Command or does it have it's own pause function?

**Reply by roasterdave** [reply]
Good to know this fixed the issue.  Depending upon how the Arduino code is written it is possible that messages are lost if they arrive while the processor is busy handling a previous one.  Thus timing becomes important.  

The Serial Command action does not have a sleep command.  Some of the other action types do have a sleep command.  The Help on the Config>> Events>> Buttons tab is up to date and shows all the available commands for each action.

**Comment by NCR25** [comment]
Thank you!
